aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLCommandProcessor.java

```diff
@@ -132,4 +130,40 @@ private void bindValueToStatement(
             );
         }
     }
+
+    /**
+     * Wraps a block of code in a sqlite transaction, ensuring that the transaction is always
+     * marked successful, even if an error is encountered. We will still throw the error, but
+     * any data saved before the error will be committed to the database.
+     *
+     * @param transactionBlock The block of code to run in the transaction.
+     * @throws DataStoreException A re-thrown error from the transaction block.
+     */
+    void runInTransactionAndSucceedOnDatastoreException(TransactionBlock transactionBlock) throws DataStoreException {
+        runInTransaction(() -> {
+            try {
+                transactionBlock.run();
+            } catch (DataStoreException exception) {
+                sqliteDatabase.setTransactionSuccessful();
```

Comment:
Since the previous implementation inserted 1 record at a time, this strategy matches the previous behavior.

In addition to this, the records that were able to be saved before the exception was encountered should be valid and would improve subsequent sync attempt speed.

Reply:
Thanks for the clarification 😃